openedx/core/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -61,6 +88,16 @@ def save_bearer_token(self, token, request, *args, **kwargs):
 
         super(EdxOAuth2Validator, self).save_bearer_token(token, request, *args, **kwargs)
 
+        is_application_restricted = RestrictedApplication.objects.filter(application=request.client).exists()
+
+        if is_application_restricted:
```

Comment:
Need to update the docstring of this method.  The current docstring refers only to overrides for the client-credentials grant.